src/app/users/users.component.ts

```diff
@@ -107,11 +108,17 @@ export class UsersComponent implements OnInit, AfterViewInit {
     const tempUser = { ...user, roles: [ ...user.roles ] };
     tempUser.roles.splice(index, 1);
     this.selectedRolesMap.set(tempUser.name, tempUser.roles);
+    if (tempUser.roles.length === 0) {
+      tempUser.isUserLocked = true ;
+    } else {
+      tempUser.isUserLocked = false ;
+    }
     delete tempUser.selected;
     this.couchService.put('_users/org.couchdb.user:' + tempUser.name, tempUser).subscribe((response) => {
       console.log('Success!');
       user.roles.splice(index, 1);
       user._rev = response.rev;
+      this.initializeData();

```

Comment:
Do not need to rerun initialize data as we are updating the necessary fields on the user documents